### PR TITLE
Logic moved to RequestService, PUT does not work yet

### PR DESCRIPTION
**Pull request by mnapa**
Moving logic from RequestController to RequestService as was recommended by Lana under https://github.com/ESI2022-team-e/backend/pull/2

### CODE REVIEW

**Comment by mnapa** [comment]
updateRequest method was fixed under this branch as well, solution suggested by Lana

File: src/main/java/esi/backend/controller/RequestController.java [hunk 1]
```diff
@@ -57,44 +57,12 @@ public void addRequest(@RequestBody Request request, @PathVariable UUID carId) {
     }
```
Comment (karoliineh) [inline src/main/java/esi/backend/controller/RequestController.java:48]: Logic from `addRequest` should also be moved to `RequestService` class.

File: src/main/java/esi/backend/service/RequestService.java [hunk 2]
```diff
@@ -16,6 +19,8 @@ public class RequestService {
```
Comment (karoliineh) [inline src/main/java/esi/backend/service/RequestService.java:23]: Unfortunately couldn't comment on unchanged lines but in method `getRequest(UUID id)` I suggest getting rid of the `Optional<Request>` return type and handling the `Optional` case in the method. The new return type could be  `ResponseEntity<Request>`. You can check my PR #7 class `InvoiceService` method `getInvoice(UUID id)` for inspo on how to get rid of `Option` (has yet to change to the `ResponseEntity` layout) and Lana's PR #6 class `RequestService` on how to get the `ResponseEntity<Request>` type.

File: src/main/java/esi/backend/service/RequestService.java [hunk 3]
```diff
@@ -35,11 +40,44 @@ public void addRequest(Request request) {
         requestRepository.save(request);
     }
 
-    public void updateRequest(Request request) {
-        requestRepository.save(request);
+    public void updateRequest(UserDetails user, Request request, UUID id) {
+        Request req = requestRepository.findById(id)
+                .orElseThrow(() -> new ResourceNotFoundException("Request with id " + id + "not found"));
```
Comment (karoliineh) [inline src/main/java/esi/backend/service/RequestService.java]:
```suggestion
                .orElseThrow(() -> new ResourceNotFoundException("Request with id " + id + " not found"));
```

**Review by karoliineh — CHANGES_REQUESTED** [review]

File: src/main/java/esi/backend/service/RequestService.java (hunk 2 above)
Comment (mnapa) [reply src/main/java/esi/backend/service/RequestService.java:23]: ResponseEntity added with [976c758](https://github.com/ESI2022-team-e/backend/pull/5/commits/976c758bb4d379e88f538fc72116b3bad85d5d9e)

File: src/main/java/esi/backend/controller/RequestController.java (hunk 1 above)
Comment (mnapa) [reply src/main/java/esi/backend/controller/RequestController.java:48]: Implemented with [423b716](https://github.com/ESI2022-team-e/backend/pull/5/commits/423b7166b579228a2cb126708abfafa765986068)

File: src/main/java/esi/backend/controller/RequestController.java [hunk 4]
```diff
@@ -23,71 +17,39 @@
...
     public ResponseEntity<?> addRequest(@AuthenticationPrincipal final UserDetails currentUser, @RequestBody Request request, @PathVariable UUID carId) {
-        Optional<Car> optionalCar = carService.getCar(carId);
-        if (optionalCar.isPresent()) {
-            request.setCar(optionalCar.get());
-        } else throw new ResourceNotFoundException("Car with id" + carId + "not found");
-        Optional<Customer> optionalCustomer = customerRepository.findByUsername(currentUser.getUsername());
-        optionalCustomer.ifPresent(request::setCustomer);
-        requestService.addRequest(request);
-        return ResponseEntity.ok("Request created successfully!");
+        requestService.addRequest(currentUser, request, carId);
```
Comment (karoliineh) [inline src/main/java/esi/backend/controller/RequestController.java:46]: `carId` is passed as the 3rd argument here, but then in the `RequestService` `updateRequest` method, it is used for retrieving the request and checking if it exists.

File: src/main/java/esi/backend/service/RequestService.java [hunk 5]
```diff
@@ -20,30 +23,61 @@ public class RequestService {
...
+        Optional<Customer> optionalCustomer = customerRepository.findByUsername(currentUser.getUsername());
+        optionalCustomer.ifPresent(request::setCustomer);
         requestRepository.save(request);
     }
 
-    public void updateRequest(Request request) {
-        requestRepository.save(request);
+    public void updateRequest(UserDetails user, Request request, UUID id) {
+        Request req = requestRepository.findById(id)
+                .orElseThrow(() -> new ResourceNotFoundException("Request with id " + id + " not found"));
```
Comment (karoliineh) [inline src/main/java/esi/backend/service/RequestService.java:61]: In InvoiceService I did so:
```
Rental rental = rentalRepository.findById(rentalId).orElse(null);
  if (rental == null) {
    new ResponseEntity<>(HttpStatus.NOT_FOUND);
    return;
  }
```
in all void methods. I think it's better for consistency as a ResponseEntity is then returned in any case.

**Review by karoliineh — APPROVED** [review]

**Pull request merged**